baseplate/url.py

```diff
@@ -0,0 +1,32 @@
+"""URL domain parsing."""
+
+
+import re
+
+DOMAIN_REGEX = re.compile("(?i)(?:.+?://)?([^/:#?]*)")
+DOMAIN_PREFIX_REGEX = re.compile(r"^www\d*\.")
+
+
+def is_subdomain(subdomain: str, base: str) -> bool:
+    """Check if a domain is equal to or a subdomain of a base domain."""
+    return subdomain == base or (subdomain is not None and subdomain.endswith("." + base))
+
+
+def get_domain(url: str) -> str:
+    """Take a URL and return the domain part (minus www.) if present."""
+    match = DOMAIN_REGEX.search(url)
+    if match:
+        domain = strip_www(match.group(1))
+    else:
+        domain = url
+    return domain.lower()
+
+
+def strip_www(domain: str) -> str:
```

Comment:
Where do we use this outside r2?

Reply:
I think it's really just used by `get_domain`?  I'm not sure about other places, in gateway we used this to check that we are sending a request to `reddit.com` before adding trace + auth + shared secret headers

Reply:
I don't really want to encourage talking manually to r2. If we're going to have more r2 comms in the future, let's do it with that r2client library and standardize that way. If things like strip_www are just for r2's sake, that seems like something we should just stay away from.